### PR TITLE
k8s: add better error handling for status 413 (entity too large)

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas, @nicksieger,

Please review the following commits I made in branch nicks/ch13123:

a253cd2eef6db8545aae8a0b023decdbb6f43b4c (2021-12-17 12:55:31 -0500)
k8s: add better error handling for status 413 (entity too large)
Fixes https://github.com/tilt-dev/tilt/issues/5279

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/k8s/client.go [hunk 1]
```diff
@@ -540,11 +542,23 @@ var MetadataAnnotationsTooLongRe = regexp.MustCompile(`metadata.annotations: Too
 // However, annotations have a max size of 256k. Large objects such as configmaps can exceed 256k, which makes
 // apply unusable, so we need to fall back to delete/create
 // https://github.com/kubernetes/kubectl/issues/712
-func maybeAnnotationsTooLong(stderr string) (string, bool) {
+//
+// We've also seen this reported differently, with a 413 HTTP error.
+// https://github.com/tilt-dev/tilt/issues/5279
+func maybeTooLargeError(err error) (string, bool) {
+	statusErr, isStatusErr := err.(*apiErrors.StatusError)
```
Comment (nicksieger) [inline internal/k8s/client.go:549]: Probably didn't notice this was already imported with auto-imports...
```suggestion
	statusErr, isStatusErr := err.(*apierrors.StatusError)
```

File: internal/k8s/client.go [hunk 2]
```diff
@@ -14,6 +15,7 @@ import (
 	"github.com/pkg/errors"
 	"helm.sh/helm/v3/pkg/kube"
 	v1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (nicksieger) [inline internal/k8s/client.go:18]: Can remove this with the change below

File: internal/k8s/client.go [hunk 3]
```diff
@@ -540,11 +542,23 @@ var MetadataAnnotationsTooLongRe = regexp.MustCompile(`metadata.annotations: Too
...
+	if isStatusErr && statusErr.ErrStatus.Code == http.StatusRequestEntityTooLarge {
+		return err.Error(), true
+	}
+
+	stderr := err.Error()
 	for _, line := range strings.Split(stderr, "\n") {
 		if MetadataAnnotationsTooLongRe.MatchString(line) {
 			return line, true
 		}
+		if strings.Contains(line, "the server responded with the status code 413") {
```
Comment (nicksieger) [inline internal/k8s/client.go:559]: sigh

**Review by nicksieger — APPROVED** [review]

File: internal/k8s/client.go (hunk 3 above)
Comment (nicks) [reply internal/k8s/client.go:561]: ya, unfortunately, a lot of the Go ecosystem doesn't do a great job preserving error types, so it's generally always good to check error messages

**Pull request merged**